doc/lsst.pipe.base/command-line-task-prov-howto.rst

```diff
@@ -0,0 +1,76 @@
+The best way to avoid configuration conflicts is to run each experimental command-line task execution with a different output :ref:`rerun <command-line-task-data-repo-using-reruns>`.
+Creating separate reruns gives you the ability to later analyze how dataset differ with task configuration.
+See :ref:`command-line-task-data-repo-using-reruns` for details on how to do this.
+
+If you cannot create a new output rerun, and must re-use an existing data repository, you can overwrite (clobber) existing task configuration metadata with the current configurations to bypass the consistency check.
+Use the :option:`--clobber-config` argument to do so:
+
+.. code-block:: bash
+
+   task.py REPOPATH --output output --id --clobber-config ...
+
+.. warning::
+
+   :option:`--clobber-config` is subject to race conditions because it is modifying a file in the Butler repository.
+   It should not be used in conjunction with parallel processing (such as :option:`-j`/:option:`--processes` or with "driver" command-line tasks).
+
+.. _command-line-task-prov-howto-versions:
+
+How to override software version checks with :option:`--no-versions` or :option:`--clobber-versions`
+====================================================================================================
+
+If a command-line task attempts to write datasets to a Butler output repository that already contains data processed with a different version of LSST Science Pipelines software, a "``Version mismatch``" error is triggered.
+This error should be heeded in most cases because datasets created by one version of the LSST Science Pipelines may not be compatible with another version.
+Developers may need to suppress this error during testing, though.
+There are two ways to do this: :option:`--no-versions`, and :option:`--clobber-versions`.
+
+Use :option:`--no-versions` to simply bypass version checking:
+
+.. code-block:: bash
+
+   task.py REPOPATH --output output --id --no-versions ...
+
+This argument is thread-safe and likely ideal for developers.
+
+An alternative method of bypassing software version provenance checks is the :option:`--clobber-versions` argument:
+
+.. code-block:: bash
+
+   task.py REPOPATH --output output --id --clobber-versions ...
+
+Like :option:`--clobber-config`, the :option:`--clobber-versions` argument replaces software version provenance information in the Butler data repository with current version information.
+This argument is useful if you wish to continue using an existing Butler data repository with a different version of the Science Pipelines, and are confident that there are not dataset incompatibilities.
+
+.. warning::
+
+   :option:`--clobber-versions` is subject to race conditions because it is modifying a file in the Butler repository.
+   It should not be used in conjunction with parallel processing (such as :option:`-j`/:option:`--processes` or with "driver" command-line tasks).
+
```

Comment:
I think you can safely run a non-existent data ID with `-j 1 --clobber-config` and then proceed with the new config and a real data ID.

It'd be good to provide a less tacky way to do this.  Or maybe simply omitting the `--id` works (or is that taken to mean all IDs as `--id` is documented to mean?  If so, it's another reason to remove that implicit globbing behaviour.